admiral/pkg/clusters/types.go

```diff
@@ -130,6 +133,168 @@ func (g *globalTrafficCache) Delete(identity string, environment string) {
 	}
 }
 
+type RoutingPolicyHandler struct {
+	RemoteRegistry *RemoteRegistry
+	ClusterID	   string
+}
+
+type routingPolicyCache struct {
+	//map of routing policies key=environment.identity, value: RoutingPolicy object
+	// only one routing policy per identity + env is allowed
+	identityCache map[string]*v1.RoutingPolicy
+
+	//// map of dependent identity + env -> [] routingPolicy. There can be potentially multiple routingPolicies that can apply to a client.
+	//dependentRpCache map[string][]*v1.RoutingPolicy
+
+	mutex *sync.Mutex
+}
+
+
+func (r *routingPolicyCache) Delete(identity string, environment string) {
+	key := common.ConstructRoutingPolicyKey(environment, identity)
+	if _, ok := r.identityCache[key]; ok {
+		log.Infof("Deleting RoutingPolicy with key=%s from global RoutingPolicy cache", key)
+		delete(r.identityCache, key)
+	}
+}
+
+func (r *routingPolicyCache ) GetFromIdentity(identity string, environment string) *v1.RoutingPolicy {
+	return r.identityCache[common.ConstructRoutingPolicyKey(environment, identity)]
+}
+
+func (r *routingPolicyCache) Put(rp *v1.RoutingPolicy) error {
+	if rp.Name == "" {
+		//no RoutingPolicy, throw error
+		return errors.New("cannot add an empty RoutingPolicy to the cache")
+	}
+	defer r.mutex.Unlock()
```

Comment:
```suggestion
         r.mutex.Lock()
	defer r.mutex.Unlock()
```

Reply:
Since the mutex lock is the next line after differ, what does your suggestion change?